### PR TITLE
pkp/pkp-lib#3578: harmonizing Form::initData() signatures

### PR DESCRIPTION
**Pull request by crism**
Fixes part of #3578; may break plugins.

### CODE REVIEW

File: classes/controllers/grid/GridHandler.inc.php [hunk 1]
```diff
@@ -876,16 +876,6 @@ protected function renderFilter($request, $filterData = array()) {
 		switch(true) {
 			case $form === null: // No filter form.
 				return '';
-			case is_a($form, 'Form'): // Form object subclass
```
Comment (asmecher) [inline classes/controllers/grid/GridHandler.inc.php]: I think this case is still used (grep around for subclasses of `ClientFormHandler`).

File: controllers/grid/settings/genre/form/GenreForm.inc.php [hunk 2]
```diff
@@ -61,9 +61,9 @@ function __construct($genreId = null) {
 	/**
 	 * Initialize form data from current settings.
 	 * @param $args array
-	 * @param $request PKPRequest
 	 */
-	function initData($args, $request) {
+	function initData($args) {
```
Comment (asmecher) [inline controllers/grid/settings/genre/form/GenreForm.inc.php]: Was `$args` intentionally left here?

File: controllers/grid/settings/submissionChecklist/SubmissionChecklistGridHandler.inc.php [hunk 3]
```diff
@@ -124,7 +124,7 @@ function editItem($args, $request) {
 		$submissionChecklistId = isset($args['rowId']) ? $args['rowId'] : null;
 		$submissionChecklistForm = new SubmissionChecklistForm($submissionChecklistId);
 
-		$submissionChecklistForm->initData($args, $request);
+		$submissionChecklistForm->initData($args);
```
Comment (asmecher) [inline controllers/grid/settings/submissionChecklist/SubmissionChecklistGridHandler.inc.php]: (As above -- not sure if `$args` is still needed)

File: classes/controllers/grid/GridHandler.inc.php (hunk 1 above)
Comment (crism) [reply classes/controllers/grid/GridHandler.inc.php]: @asmecher, pointers? I don’t see `ClientFormHandler` anywhere in the code base, except in JavaScript. I looked pretty thoroughly at usage before deleting that clause.

File: controllers/grid/settings/genre/form/GenreForm.inc.php (hunk 2 above)
Comment (crism) [reply controllers/grid/settings/genre/form/GenreForm.inc.php]: Yes; it’s referenced at the end of the body of that method. There may be a better way to pass it in, but this satisfies PHP’s warnings.

File: controllers/grid/settings/submissionChecklist/SubmissionChecklistGridHandler.inc.php (hunk 3 above)
Comment (crism) [reply controllers/grid/settings/submissionChecklist/SubmissionChecklistGridHandler.inc.php]: As above; `SubmissionChecklistForm` relies on a few things being passed in.

File: classes/controllers/grid/GridHandler.inc.php (hunk 1 above)
Comment (asmecher) [reply classes/controllers/grid/GridHandler.inc.php]: Just grepped a little further and it looks like you're right. I think the submission lists used to use this case but they've been adapted to vue.js. However, I think you still need to change the self-doc for GridHandler::getFilterForm.

**Pull request merged**